File: app/src/main/java/ru/geekbrains/notes/domain/observer/Publisher.java
package ru.geekbrains.notes.domain.observer;

import java.util.ArrayList;
import java.util.List;

import ru.geekbrains.notes.domain.note.Note;

// Обработчик подписок
public class Publisher {

    private List<ObserverNote> observers;

    //Конструктор
    public Publisher() {
        observers = new ArrayList<>();
    }

    // Подписать
    public void subscribe(ObserverNote observer) {
        observers.add(observer);
    }

    // Отписать
    public void unsubscribe(ObserverNote observer) {
        observers.remove(observer);
    }

    // Разослать событие
    public void notify(Note note) {
        for (ObserverNote observer: observers) {
            observer.updateNote(note);
        }
    }
}

File: app/src/main/java/ru/geekbrains/notes/observer/Publisher.kt
package ru.geekbrains.notes.observer

import java.util.*

// Обработчик подписок
class Publisher {
    private val observers: MutableList<ObserverNote>

    // Подписать
    fun subscribe(observer: ObserverNote) {
        observers.add(observer)
    }

    // Отписать
    fun unsubscribe(observer: ObserverNote) {
        observers.remove(observer)
    }

    // Разослать событие
    fun notify(noteID: Int, typeEvent: Int) {
        for (observer in observers) {
            observer.updateNote(noteID, typeEvent)
        }
    }

    //Конструктор
    init {
        observers = ArrayList()
    }
}